machine/v1/types_nutanixprovider.go

```diff
@@ -154,6 +165,139 @@ type NutanixResourceIdentifier struct {
+	// type is the identifier type to use for this GPU resource.
+	// The GPU device can be identified either by name or deviceID.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Type NutanixGPUIdentifierType `json:"type"`
+
+	// deviceID is the GPU device ID with the integer value.
+	// +optional
+	// +unionMember
+	DeviceID *int32 `json:"deviceID,omitempty"`
+
+	// name is the GPU device name
+	// +optional
+	// +unionMember
+	Name *string `json:"name,omitempty"`
+}
+
+// NutanixDiskFlashMode is an enumeration of different disk flash modes.
+// +kubebuilder:validation:Enum=On;Off
+type NutanixDiskFlashMode string
+
+const (
+	// NutanixDiskFlashModeOn represents the disk flash mode is On.
+	NutanixDiskFlashModeOn NutanixDiskFlashMode = "On"
+
+	// NutanixDiskFlashModeOff represents the disk flash mode is Off.
+	NutanixDiskFlashModeOff NutanixDiskFlashMode = "Off"
+)
+
+// NutanixVMStorageConfig specifies the storage configuration parameters for VM disks.
+type NutanixVMStorageConfig struct {
+	// flashMode specifies whether to pin the VM disk to the flash tier.
+	// +kubebuilder:default=Off
+	FlashMode NutanixDiskFlashMode `json:"flashMode"`
+
+	// storageContainer refers to the storage_container used by the VM disk.
+	// Currently the storageContainer configuration is supported only by using its UUID.
+	// In the future, we may support to identify storageContainer by its name.
+	// +optional
+	StorageContainer *NutanixResourceIdentifier `json:"storageContainer"`
+}
+
+// NutanixDiskDeviceType is the VM disk device type.
+// +kubebuilder:validation:Enum=Disk;CDRom
+type NutanixDiskDeviceType string
+
+const (
+	// NutanixDiskDeviceTypeDisk specifies the VM disk device type is DISK.
+	NutanixDiskDeviceTypeDisk NutanixDiskDeviceType = "Disk"
+
+	// NutanixDiskDeviceTypeCDROM specifies the VM disk device type is CDROM.
+	NutanixDiskDeviceTypeCDROM NutanixDiskDeviceType = "CDRom"
+)
+
+// NutanixDiskAdapterType is an enumeration of different disk device adapter types.
+// +kubebuilder:validation:Enum:=SCSI;IDE;PCI;SATA;SPAPR
+type NutanixDiskAdapterType string
+
+const (
+	// NutanixDiskAdapterTypeSCSI represents the disk adapter type "SCSI".
+	NutanixDiskAdapterTypeSCSI NutanixDiskAdapterType = "SCSI"
+
+	// NutanixDiskAdapterTypeIDE represents the disk adapter type "IDE".
+	NutanixDiskAdapterTypeIDE NutanixDiskAdapterType = "IDE"
+
+	// NutanixDiskAdapterTypePCI represents the disk adapter type "PCI".
+	NutanixDiskAdapterTypePCI NutanixDiskAdapterType = "PCI"
+
+	// NutanixDiskAdapterTypeSATA represents the disk adapter type "SATA".
+	NutanixDiskAdapterTypeSATA NutanixDiskAdapterType = "SATA"
+
+	// NutanixDiskAdapterTypeSPAPR represents the disk adapter type "SPAPR".
+	NutanixDiskAdapterTypeSPAPR NutanixDiskAdapterType = "SPAPR"
+)
+
+// NutanixVMDiskDeviceProperties specifies the disk device type and disk address.
+type NutanixVMDiskDeviceProperties struct {
+	// deviceType is a disk type. The default is DISK.
```

Comment:
DISK is not correct anymore, will want to add a list of supported values as I demonstrated on the type field earlier

Reply:
changed